src/components/ResourceBrowser/ResourceList/AdminTerminal.tsx

```diff
@@ -20,27 +20,40 @@ import { Progressing, ErrorScreenManager, Reload, useAsync } from '@devtron-labs
 import ClusterTerminal from '../../ClusterNodes/ClusterTerminal'
 import { createGroupSelectList, filterImageList } from '../../common'
 import { createTaintsList } from '../../cluster/cluster.util'
-import { getClusterList, clusterNamespaceList } from '../../ClusterNodes/clusterNodes.service'
+import { clusterNamespaceList, getClusterCapacity } from '../../ClusterNodes/clusterNodes.service'
 import { getHostURLConfiguration } from '../../../services/service'
 import { AdminTerminalProps, URLParams } from '../Types'
 
-const AdminTerminal: React.FC<AdminTerminalProps> = ({ isSuperAdmin, updateTerminalTabUrl }: AdminTerminalProps) => {
+const AdminTerminal: React.FC<AdminTerminalProps> = ({
+    isSuperAdmin,
+    clusterName,
+    updateTerminalTabUrl,
+}: AdminTerminalProps) => {
     const { clusterId } = useParams<URLParams>()
 
     const [loading, data, error] = useAsync(() =>
-        Promise.all([getClusterList(), getHostURLConfiguration('DEFAULT_TERMINAL_IMAGE_LIST'), clusterNamespaceList()]),
+        Promise.all([
+            getClusterCapacity(clusterId),
+            getHostURLConfiguration('DEFAULT_TERMINAL_IMAGE_LIST'),
+            clusterNamespaceList(),
+        ]),
     )
 
-    const [detailClusterList = null, hostUrlConfig = null, _namespaceList = null] = data || []
+    const [clusterDetail = null, hostUrlConfig = null, _namespaceList = null] = data || []
 
-    const selectedDetailsCluster = useMemo(
-        () => detailClusterList?.result.find((cluster) => cluster.id === +clusterId) || null,
-        [detailClusterList, clusterId],
-    )
+    const details = useMemo(() => {
+        const detail = clusterDetail?.result ?? null
+
+        const imageList = JSON.parse(hostUrlConfig?.result.value ?? null)
+        const clusterImageList = filterImageList(imageList, detail?.serverVersion ?? '')
+
+        const nodeGroups = createGroupSelectList(detail?.nodeDetails, 'nodeName')
+        const taints = detail ? createTaintsList(detail.nodeDetails, 'nodeName') : null
 
-    const imageList = useMemo(() => JSON.parse(hostUrlConfig?.result.value || null), [hostUrlConfig])
+        return { ...(detail ?? {}), nodeGroups, taints, clusterImageList }
+    }, [clusterDetail, clusterId])
```

Comment:
Why clusterId dependency is required here?

Reply:
Removed those dependencies and now the only dependency is data from the useAsync